### ISSUE TITLE
Error: Python module keras was not found.

### ISSUE THREAD

**Issue by MrAlecTricity** [rstudio/keras#639]
I have looked at other similar issues but have not found a solution to my specific case, so I opened a new one.

I have a problem running Keras in R. When running `mnist <- dataset_mnist()` I get the error `Error: ModuleNotFoundError: No module named 'keras'`

I have installed the most recent version of Anaconda, I am running R version 3.5.1 in RStudio 1.1.463. In R I have run
```
library("keras")
install_keras()
```
and this succeeds without errors.

When trying to run `mnist <- dataset_mnist()` I initially got the following error message:
```
ImportError: No module named keras
Use the install_keras() function to install the core Keras library
Error: Error loading Python module keras
```
When running
```
library(reticulate)
py_config()
```
I got the following output:
```
python:         /usr/bin/python
libpython:      /System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/config/libpython2.7.dylib
pythonhome:     /System/Library/Frameworks/Python.framework/Versions/2.7:/System/Library/Frameworks/Python.framework/Versions/2.7
version:        2.7.10 (default, Oct  6 2017, 22:29:07)  [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.31)]
numpy:          /System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/numpy
numpy_version:  1.8.0
keras:          [NOT FOUND]
```
After looking at suggested solutions to other issues, I added this line to my .Rprofile
```
Sys.setenv(RETICULATE_PYTHON = "/anaconda3/envs/r-tensorflow/bin")
```
However when I run mnist <- dataset_mnist() now I get the following error:
```
Error: Python module keras was not found.

Detected Python configuration:

python:         /anaconda3/envs/r-tensorflow/bin/python
libpython:      /anaconda3/envs/r-tensorflow/lib/libpython3.6m.dylib
pythonhome:     /anaconda3/envs/r-tensorflow:/anaconda3/envs/r-tensorflow
version:        3.6.7 | packaged by conda-forge | (default, Nov 20 2018, 18:37:09)  [GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)]
numpy:           [NOT FOUND]
keras:          /anaconda3/envs/r-tensorflow/lib/python3.6/site-packages/keras

NOTE: Python version was forced by RETICULATE_PYTHON
```
Does anyone know what I can do to solve this?

**Comment by ezgraphs** [comment]
I hit a problem running install_keras() on a OSX machine where I had previously installed anaconda3 (but was seeing an error - don't recall details at this point).  It seems like there are a variety of possible python configurations that result in related problems.

* The solution that worked for me involved installing keras outside of RStudio using anaconda:  https://github.com/tensorflow/tensorflow/issues/6999#issuecomment-278459224
* Related discussion:  https://github.com/rstudio/keras/issues/102

**Comment by skeydan** [comment]
@MrAlecTricity did the procedure suggested by @ezgraphs work for you?

If not, can you provide the output of

```
library(reticulate)
sys <- import("sys")
sys$path
import("keras")
```

?

**Comment by brianjstroh** [comment]
@MrAlecTricity I was running into this same problem after installing keras outside of RStudio using Anaconda (installing through RStudio was blocked by my company's system restrictions). Turns out all I had to do was restart the R session and everything worked fine after than. Hope this helps!

**Comment by skeydan** [comment]
Hello to everybody on this thread,

if you still have problems using `install_tensorflow()` from RStudio, please try again now.

First install `tensorflow` directly from GitHub, as in 

```
devtools::install_github("rstudio/tensorflow")
devtools::install_github("rstudio/keras")
```

Then do 

```
tensorflow::install_tensorflow()
tensorflow::tf_config()
```

which should give you version 1.12.

We have been installing TF 1.10 until yesterday because of a bug in that will only be fixed in TF 1.13 (which should be out anytime but unfortunately isn't yet). Now with people having installation problems due to incompatibilities with `numpy` as installed by `conda`, we decided to switch to TF 1.12 now and as soon as TF 1.13 is actually available, we'll install that by default.

Thanks for reporting the problem!

If you still run into problems installing, please open another issue and describe the problem, indicating the output of  

```
reticulate::py_discover_config()
reticulate::use_condaenv("r-tensorflow")
reticulate::py_config()
```

 Thanks!

**Comment by Steviey** [comment]
+1

**Comment by dkxmen** [comment]
> tensorflow::install_tensorflow()
> tensorflow::tf_config()

Thanks, this worked, but the version is "**TensorFlow v1.13.1"** on 19th May 2019.
